src/main/java/iguanaman/iguanatweakstconstruct/claybuckets/ClayBucketHandler.java

```diff
@@ -4,16 +4,41 @@
 import cpw.mods.fml.common.eventhandler.EventPriority;
 import cpw.mods.fml.common.eventhandler.SubscribeEvent;
 import net.minecraft.block.Block;
+import net.minecraft.block.BlockCauldron;
 import net.minecraft.entity.passive.EntityCow;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraftforge.event.entity.player.EntityInteractEvent;
 import net.minecraftforge.event.entity.player.FillBucketEvent;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
 import tconstruct.smeltery.TinkerSmeltery;
 
 public class ClayBucketHandler {
+    @SubscribeEvent
+    public void EntityInteract(PlayerInteractEvent event) {
+        Block block = event.world.getBlock(event.x, event.y, event.z);
+        EntityPlayer player = event.entityPlayer;
+        ItemStack equipped = player.getCurrentEquippedItem();
+
+        if(block == null
```

Comment:
First of all block nullability is covered by `block instanceof BlockCauldron`

AFAIK: block can't be null